File: linux/fluera_engine_plugin.cc
// fluera_engine_plugin.cc — Main Flutter Linux plugin entry point
#include "include/fluera_engine/fluera_engine_plugin.h"
#include "gl_stroke_overlay_plugin.h"

void fluera_engine_plugin_register_with_registrar(
    FlPluginRegistrar *registrar) {
  gl_stroke_overlay_plugin_register_with_registrar(registrar);
}

File: macos/Classes/FlueraEnginePlugin.swift
import FlutterMacOS

/// 🚀 FlueraEnginePlugin — Main entry point for Fluera Engine's native macOS capabilities.
///
/// Registers the Metal stroke overlay plugin for low-latency GPU rendering.
public class FlueraEnginePlugin: NSObject, FlutterPlugin {
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        MetalStrokeOverlayPlugin.register(with: registrar)
    }
}
